tests/console/sssd_openldap_functional.pm

```diff
@@ -0,0 +1,105 @@
+#
+# 1. nss_sss test: look up user identity with id: uid and gid
+# 2. pam_sss test: ssh login localhost as remote user.
+# 3. write permission test: change remote user password with passwd
+# 4. sssd-sudo test: Sudo run command as another remote user with sudoers rules defined in server
+# 5. offline test: shutdown server, run test cases above again
+#
+# Detailed testcases: https://bugzilla.suse.com/tr_show_case.cgi?case_id=1768711
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package sssd_openldap_functional;
+use base 'consoletest';
+use testapi;
+use strict;
+use warnings;
+use utils;
+use version_utils;
+use registration qw(add_suseconnect_product get_addon_fullname);
+
+sub run {
+    select_console("root-console");
+    if (!get_var("SSSD_OPENLDAP_FUNCTIONAL_SETUP")) {
+        if (is_sle) {
+            add_suseconnect_product('PackageHub', undef, undef, undef, 300, 1);
+            is_sle('<15') ? add_suseconnect_product("sle-module-containers", 12) : add_suseconnect_product("sle-module-containers");
+        }
+        zypper_call("in sssd sssd-ldap openldap2-client sshpass docker");
+        systemctl('enable --now docker');
+        #Select container base image by specifying variable BASE_IMAGE_TAG. (for sles using sle15sp3 by default)
+        my $pkgs = "openldap2 sudo";
+        my $tag  = get_var("BASE_IMAGE_TAG");
+        unless ($tag) {
+            if (is_opensuse) { $tag = (is_tumbleweed) ? "opensuse/tumbleweed" : "opensuse/leap";
+            } else           { $tag = "registry.suse.com/suse/sle15:15.3"; }
+        }
+        # build container
+        # build image, create container, setup openldap database and import testing data
+        assert_script_run("mkdir /tmp/sssd && cd /tmp/sssd");
+        assert_script_run("curl -s " . "--remote-name-all " . data_url('sssd/openldap/{user.ldif,slapd.conf,Dockerfile}'));
+        assert_script_run("curl -s " . "--remote-name-all " . data_url('sssd/openldap/ldapserver.{key,crt,csr}'));
+        assert_script_run(qq(docker build -t openldap2_image --build-arg tag="$tag" --build-arg pkgs="$pkgs" .), timeout => 600);
+        assert_script_run('docker run -itd --name ldap_container --hostname ldapserver --restart=always openldap2_image');
+        assert_script_run("docker exec ldap_container sed -n '/ldapserver/p' /etc/hosts >> /etc/hosts");
+
+        # Configure sssd on the host
+        assert_script_run("curl -s " . data_url("sssd/openldap/sssd.conf") . " -o /etc/sssd/sssd.conf");
+        assert_script_run("curl -s " . data_url("sssd/openldap/nsswitch.conf") . " -o /etc/nsswitch.conf");
+        assert_script_run("curl -s " . data_url("sssd/openldap/ldapserver.crt") . " -o /etc/sssd/ldapserver.crt");
+        assert_script_run("curl -s " . data_url("sssd/openldap/config") . " --create-dirs -o ~/.ssh/config");
+        #    systemctl("stop nscd.service");
+        systemctl('disable --now nscd.service');
+
+        #    disable_and_stop_service("nscd");
```

Comment:
disable_and_stop_service("nscd"); ?

Reply:
yes, "enable --now" will do that, "stop and disable'.  It's one line code. disable_and_stop_service is multiple lines. So I am using "enable --now". I just undated code and removed the comments.

Reply:
I mean in comment you had stop and disable **nscd** but command was enabling **sssd**.

Reply:
.  Line 68 and 71 were not comments.  They were source code which were commented out by me :). I forgot to deleted them.